cli/provision.py

```diff
@@ -238,9 +183,14 @@ def run(self):
 
     def collect_settings(self):
         settings_files = self.sub_command.get_settings_files()
-        settings_dict = self.sub_command.get_settings_dict()
+        arguments_dict = {APPLICATION: spec.get_arguments_dict(self.args)}
```

Comment:
I think the get_arguments_dict should be moved out of spec file. Application should transform spec to the dict. Spec.py should don't care about required format for application.